src/com/owncloud/android/providers/UsersAndGroupsSearchProvider.java

```diff
@@ -148,31 +172,50 @@ private Cursor searchForUsersOrGroups(Uri uri) {
         if (names.size() > 0) {
             response = new MatrixCursor(COLUMNS);
             Iterator<JSONObject> namesIt = names.iterator();
-            int count = 0;
             JSONObject item;
-            String displayName;
-            Uri dataUri;
-            Uri userBaseUri = new Uri.Builder().scheme("content").authority(DATA_USER).build();
-            Uri groupBaseUri = new Uri.Builder().scheme("content").authority(DATA_GROUP).build();
+            String displayName = null;
+            Uri dataUri = null;
+            int count = 0;
+
+            Uri userBaseUri = new Uri.Builder().scheme(CONTENT).authority(DATA_USER).build();
+            Uri groupBaseUri = new Uri.Builder().scheme(CONTENT).authority(DATA_GROUP).build();
+            Uri remoteBaseUri = new Uri.Builder().scheme(CONTENT).authority(DATA_REMOTE).build();
+
+            FileDataStorageManager manager = new FileDataStorageManager(account, getContext().getContentResolver());
+            boolean federatedShareAllowed = manager.getCapability(account.name).getFilesSharingFederationOutgoing()
+                    .isTrue();
+
             try {
                 while (namesIt.hasNext()) {
                     item = namesIt.next();
                     String userName = item.getString(GetRemoteShareesOperation.PROPERTY_LABEL);
                     JSONObject value = item.getJSONObject(GetRemoteShareesOperation.NODE_VALUE);
-                    byte type = (byte) value.getInt(GetRemoteShareesOperation.PROPERTY_SHARE_TYPE);
+                    int type = value.getInt(GetRemoteShareesOperation.PROPERTY_SHARE_TYPE);
                     String shareWith = value.getString(GetRemoteShareesOperation.PROPERTY_SHARE_WITH);
-                    if (GetRemoteShareesOperation.GROUP_TYPE.equals(type)) {
+
+                    if (ShareType.GROUP.getValue() == type) {
                         displayName = getContext().getString(R.string.share_group_clarification, userName);
                         dataUri = Uri.withAppendedPath(groupBaseUri, shareWith);
-                    } else {
+                    } else if (ShareType.FEDERATED.getValue() == type && federatedShareAllowed) {
+                        if (userName.equals(shareWith)) {
+                            displayName = getContext().getString(R.string.share_remote_clarification, userName);
+                        } else {
+                            displayName = getContext().getString(R.string.share_known_remote_clarification, userName,
+                                    shareWith.split("@")[1]);
```

Comment:
Shouldn't use the element in position ```[1]```,  but the last element, not assuming that there will be only 2. In some servers "@" may be a valid character in the username.